src/main/java/carrierManagementSystem/CarrierManagementSystemDirector.java

```diff
@@ -0,0 +1,74 @@
+package carrierManagementSystem;
+
+import java.util.Stack;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.EmptyStackException;
+import java.util.List;
+
+/**
+ * This singleton class is responsible to create LTE carrier and WCDMA carrier. 
+ * We can call this class director as it will direct the creation command from
+ * client(CarrierManagementSystemClient) to Carrier builder class. This class
+ * will also assign a random unique carrier ID (in range 1 - 1000) to each
+ * carrier it creates.
+ */
+public class CarrierManagementSystemDirector implements CarrierManagementIf {
+    private static volatile CarrierManagementSystemDirector UNIQUE_INSTANCE = null;
+    private Stack<Integer> carrierIdGenerator = new Stack<Integer>();
+    private ArrayList<Integer> randomizeCarrierId = new ArrayList<Integer>();
+
+    private CarrierManagementSystemDirector() {
+        for (int i = 1; i < 1001; i++) {
```

Comment:
Minor: Can we make `1001` some constant somewhere, instead of a magic value here? I'd also suggest making it `1000` (and adjust the conditional appropriately) since the logic implies there are a max of `1000` carriers.

Reply:
done